Turkish Language Localization

LychALPHA opened the pull request:
Here there is Turkish Localization. No map names, game modes and map descriptions changed but rather everything besides translated and localizated. There will be opening screen translation too, yet i need to learn photoshop or something for that.
Needs support for turkish charecters!
Some words overlap in UI, i don really know how to fix that.

LychALPHA committed:
Add files via upload

LychALPHA committed:
turkish localizationnn

LychALPHA committed:
Delete Translation.ini

LychALPHA committed:
Add files via upload

SadPencil commented:
I'll mark this PR as a draft since you said it still needs some work

> Needs support for turkish characters!

Can you provide a Unicode code point range? e.g., something like U+4E00 -- U+9FA5

> Some words overlap in UI

The translation system supports *slightly* adjusting UI 
You can refer to the document here https://github.com/CnCNet/xna-cncnet-client/blob/develop/Docs/Translation.md

SadPencil commented on package/Resources/Translations/tr/Translation.ini:
```diff
@@ -0,0 +1,1735 @@
+[General]
+Author=LychALPHA
+MapEncoding=utf-8
```
You can just remove this line `MapEncoding=utf-8` since it's the default option.

`MapEncoding` key is only used for a special case, e.g. tiberian sun.

SadPencil commented on package/Resources/Translations/tr/Translation.ini:
```diff
@@ -0,0 +1,1735 @@
+INI:Hotkeys:View3:UIName=3. Yer İşaretini Görüntüle 
+INI:Hotkeys:View4:Description=Kamerayı yer imi 4 üzerinde ortalayın.
+INI:Hotkeys:View4:UIName=4. Yer İşaretini Görüntüle 
+INI:Maps:Maps\Yuri's Revenge\2_across_the_frost:Description=[2] Across the Frost (PreCaptured)
```
It's okay to leave these strings untranslated. In this case, please comment out or delete these lines.

LychALPHA replied:
the whole hotkeys translation?

SadPencil replied:
> the whole hotkeys translation?

I commented on line 1076

LychALPHA replied:
oh the map list, alr

LychALPHA commented:
> I'll mark this PR as a draft since you said it still needs some work
> 
> > Needs support for turkish characters!
> 
> Can you provide a Unicode code point range? e.g., something like U+4E00 -- U+9FA5
> 
> > Some words overlap in UI
> 
> The translation system supports _slightly_ adjusting UI You can refer to the document here https://github.com/CnCNet/xna-cncnet-client/blob/develop/Docs/Translation.md




ğ (Latin Small Letter G with Breve): U+011F

Ğ (Latin Capital Letter G with Breve): U+011E

ı (Latin Small Letter Dotless I): U+0131

İ (Latin Capital Letter I with Dot Above): U+0130

ş (Latin Small Letter S with Cedilla): U+015F

Ş (Latin Capital Letter S with Cedilla): U+015E

(Latin Extended A)

LychALPHA committed:
Add files via upload

Latest version, added loading screen translations and troubleshooting for translate. Rephrased some words.

LychALPHA commented:
there is nothing more to do with Turkish localization, everything -literally- translated, loading screen images translated. Thanks to the Remnant, Turkish charecter support is on the way. Whenever new spritefont releases, this localization is ready to go.

MahBoiDeveloper commented:
@LychALPHA We've recently updated the client font, please try it to check if all letters is OK.

LychALPHA commented:
> @LychALPHA We've recently updated the client font, please try it to check if all letters is OK.

is there any way that i could check the sprite font? if you guys added latin extended A, there should be no problem. (Those letters that i listed above) And this translation ready to go.

MahBoiDeveloper commented:
> is there any way that i could check the sprite font?

Only if you replace some text in client that could be easily find.

LychALPHA commented:
> > is there any way that i could check the sprite font?
> 
> Only if you replace some text in client that could be easily find.

Checked, fonts works fine. There is no other issue. Translation ready to go.

MahBoiDeveloper commented:
@LychALPHA Please, update client translation with latest changes

LychALPHA committed:
Latest version 

Every line translated (including new lobby lines, hotkeys, missions and mission descriptions, minor translation fixes) excluding map names and map desciptions.

LychALPHA committed:
Update Translation.ini

LychALPHA committed:
Update Translation.ini

LychALPHA committed:
Latest translation by 19.06.2006

Latest translation by 19.06.2006

MahBoiDeveloper commented on Translation.ini:
Please remove this file

MahBoiDeveloper commented on tr/Translation.ini:
Please remove `tr` folder from the root of the repository

MahBoiDeveloper commented on package/Resources/Translations/tr/Translation.ini:
Campaign text doesn't translated.

<img width="1428" height="811" alt="Image" src="https://github.com/user-attachments/assets/0a296dbd-d50a-44e7-9273-172326008d75" />

LychALPHA committed:
Delete tr/launcherupdater.png

LychALPHA committed:
Delete tr directory

LychALPHA committed:
Delete Translation.ini

LychALPHA committed:
Update Translation.ini

MahBoiDeveloper commented on package/Resources/Translations/tr/Translation.ini:
```diff
@@ -0,0 +1,2112 @@
+[General]
+Author=LychALPHA
+MapEncoding=utf-8
```
```suggestion
Name=Turkish
Author=LychALPHA
```

MahBoiDeveloper committed:
Apply suggestion from @MahBoiDeveloper

LychALPHA committed:
Update Translation.ini

LychALPHA committed:
Update Translation.ini

MahBoiDeveloper reviewed (approved):
LGTM

SadPencil commented:
No one merges it? Any objection before the merge? @MahBoiDeveloper @brichardson1991 @CnCRAZER Otherwise I suggest merge it

MahBoiDeveloper commented:
> No one merges it? Any objection before the merge? @MahBoiDeveloper @brichardson1991 @CnCRAZER Otherwise I suggest merge it

There is a classic branch issue. The all changes has been made in develop instead of different branch, so the author should reopen PR with feature branch instead of their own develop

MahBoiDeveloper reviewed (changes requested):
Please, change your branch from `develop` for different one, for example `feature/turkish`. If you have a problem with using git, check official tutorials https://git-scm.com/docs/gittutorial

SadPencil commented:
> > No one merges it? Any objection before the merge? @MahBoiDeveloper @brichardson1991 @CnCRAZER Otherwise I suggest merge it
> 
> There is a classic branch issue. The all changes has been made in develop instead of different branch, so the author should reopen PR with feature branch instead of their own develop

uh, right, but this is not mandatory for a person who is just a translator and aren't very familiar with programming?

brichardson1991 commented:
Good with me i can't translate it so if it's good for others i'm fine with it.

MahBoiDeveloper commented:
> > > No one merges it? Any objection before the merge? @MahBoiDeveloper @brichardson1991 @CnCRAZER Otherwise I suggest merge it
> > 
> > 
> > There is a classic branch issue. The all changes has been made in develop instead of different branch, so the author should reopen PR with feature branch instead of their own develop
> 
> uh, right, but this is not mandatory for a person who is just a translator and aren't very familiar with programming?

btw the github cannot decide if there is a merge conflict or not. I've watched for the button state about 2 minutes and it didn't turned green.

SadPencil commented:
> > > > No one merges it? Any objection before the merge? @MahBoiDeveloper @brichardson1991 @CnCRAZER Otherwise I suggest merge it
> > > 
> > > 
> > > There is a classic branch issue. The all changes has been made in develop instead of different branch, so the author should reopen PR with feature branch instead of their own develop
> > 
> > 
> > uh, right, but this is not mandatory for a person who is just a translator and aren't very familiar with programming?
> 
> btw the github cannot decide if there is a merge conflict or not. I've watched for the button state about 2 minutes and it didn't turned green.

because there is not a single approval review comment

SadPencil commented:
@MahBoiDeveloper okay, now the button is green. There is no conflict. Do you still want to block this PR or do you want a squash merge?

MahBoiDeveloper commented:
> @MahBoiDeveloper okay, now the button is green. There is no conflict. Do you still want to block this PR or do you want a squash merge?

No, if now is ok, let's merge it
